.github/actions/assign-triage-role/assign_triage_role/triage_role_assigner.py

```diff
@@ -0,0 +1,342 @@
+		"""
+		Returns a dict of Pull Requests, associated by committer, within the last
+		:var self.since_day: days of :var self.today:.
+		"""
+		# Search for PRs and Issues on the parent repo if running on a fork
+		repo_name = self.repo.full_name if self.repo.parent is None else self.repo.parent.full_name
+
+		query = (f"repo:{repo_name} is:issue linked:pr is:closed closed:"
+		         f"{self.since_day()}..{self.today}")
+		linked_issues = self.search_issues(query=query)
+		prs_by_contributor: dict[NamedUser, list[(Issue, Issue)]] = {}
+		for linked_issue in linked_issues:
+			timeline = linked_issue.get_timeline()
+			pull_request: Optional[Issue] = None
+			for event in timeline:
+				if event.id is not None:
+					continue
+				if event.event != GH_TIMELINE_EVENT_TYPE_CROSS_REFERENCE:
+					continue
+				pr_text = event.raw_data["source"]["issue"]
+				if "pull_request" not in pr_text:
+					continue
+				pull_request = Issue(self.repo.__getattribute__("_requester"), event.raw_headers,
+					pr_text, completed=True)
+			# Do not break, in case the Issue has ever been linked to more than 1 PR in the past
+			if pull_request is None:
+				raise Exception(
+					f"Unable to find a linked Pull Request for Issue {self.repo.full_name}#{linked_issue.number}")
+			# Skip unmerged PRs
+			if "merged_at" not in pull_request.pull_request.raw_data:
+				continue
+			author = pull_request.user
+			if author.login in committers:
+				continue
+			if author not in prs_by_contributor:
+				prs_by_contributor[author] = []
+			prs_by_contributor[author].append((pull_request, linked_issue))
+		return prs_by_contributor
+
+	def ones_who_meet_threshold(self, prs_by_contributor: dict[NamedUser,
+	list[(Issue, Issue)]]) -> dict[str, list[(Issue, Issue)]]:
+		"""
+		Returns a dict of contributors who had at least self.minimum_commits Issue-closing Pull
+		Requests merged in the past self.since_days_ago days
+		"""
+		return {
+			# use only the username as the dict key
+			contributor.login: pull_requests
+			# sort contributors by commit count
+			for contributor, pull_requests in sorted(
+				prs_by_contributor.items(),
+				key=lambda item: len(item[1]),
+				# highest commit count first
+				reverse=True)
+			# only include contributors who had at least self.minimum_commits Issue-closing Pull
+			# Requests merged in the past self.since_days_ago days
+			if len(pull_requests) >= self.minimum_commits
+		}
+
+	def set_collaborators_in_asf_yaml(self, prs_by_contributor: dict[str, list[(Issue, Issue)]],
+			description: str) -> None:
+		"""
+		Writes the list of collaborators to .asf.yaml
+		"""
+		collaborators: list[str] = list(prs_by_contributor)
+		with open(ASF_YAML_FILE, encoding="utf-8") as stream:
+			github_key: Final[str] = "github"
+			collaborators_key: Final[str] = "collaborators"
+			try:
+				asf_yaml: dict[str, dict] = yaml.safe_load(stream)
+			except YAMLError as e:
+				print(f"Could not load YAML file {ASF_YAML_FILE}: {e}")
+				sys.exit(1)
```

Comment:
should this be handled in __main__ like the other exception?

Reply:
Handling `YAMLError` in `__main__` in f5207efd64